File: sdk-kotlin/src/main/kotlin/org/sinou/pydia/sdk/api/callbacks/RegistryItemHandler.java
package org.sinou.pydia.sdk.api.callbacks;

import org.sinou.pydia.sdk.api.ui.Plugin;

import java.util.Properties;

public abstract class RegistryItemHandler {

    public void onPref(String name, String value) {
    }

    public void onAction(String action, String read, String write) {
    }

    public void onWorkspace(Properties p) {
    }

    public void onPlugin(Plugin p) {
    }
}

File: sdk-kotlin/src/main/kotlin/org/sinou/pydia/sdk/client/model/parser/RegistryItemHandler.kt
package org.sinou.pydia.sdk.client.model.parser

import org.sinou.pydia.sdk.api.ui.Plugin
import java.util.Properties

abstract class RegistryItemHandler {
    fun onPref(name: String, value: String?) {}
    fun onAction(action: String, read: String?, write: String?) {}
    fun onWorkspace(p: Properties) {}
    fun onPlugin(p: Plugin) {}
}